jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/AbsoluteUnixPath.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.filesystem;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.nio.file.Path;
+import java.util.StringJoiner;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Represents a Unix-style path in absolute form (containing all path components relative to the
+ * file system root {@code /}).
+ *
+ * <p>This class is immutable and thread-safe.
+ */
+@Immutable
+public class AbsoluteUnixPath {
+
+  /**
+   * Gets a new {@link AbsoluteUnixPath} from a Unix-style path string. The path must begin with a
+   * forward slash ({@code /}).
+   *
+   * @param unixPath the Unix-style path string in absolute form
+   * @return a new {@link AbsoluteUnixPath}
+   */
+  public static AbsoluteUnixPath get(String unixPath) {
+    Preconditions.checkArgument(
+        unixPath.startsWith("/"), "Path does not start with forward slash (/): " + unixPath);
+
+    return new AbsoluteUnixPath(UnixPathParser.parse(unixPath));
+  }
+
+  /**
+   * Gets a new {@link AbsoluteUnixPath} from a {@link Path}. The {@code path} must be absolute
+   * (indicated by a non-null {@link Path#getRoot}).
+   *
+   * @param path the absolute {@link Path} to convert to an {@link AbsoluteUnixPath}.
+   * @return a new {@link AbsoluteUnixPath}
+   */
+  private static AbsoluteUnixPath fromPath(Path path) {
```

Comment:
No one seems to use this `private` method.

Reply:
Oops meant to make this public. This will be used by LayerEntry to turn the source file path into an AbsoluteUnixPath for serialization.

Reply:
I actually had a question that I was about to ask.

The source file path of `LayerEntry` is, e.g., the path of an actual file on a local machine, right? That is, it could be like "D:\my-proj\build\classes\..." on Windows. And I see we are converting it to a Unix path for serialization. (Will be `AbsoluteUnixPath` after this PR.) Doesn't it discard the drive letter (D:\)?

Reply:
Yea, that makes me think we might just need to store the windows form path for the source file in the cache metadata and remove this method.

Reply:
This is the right conversation. File a new issue for https://github.com/GoogleContainerTools/jib/pull/995#discussion_r218589436?

Reply:
Ah yes, filed #1008